### PR TITLE
Math magicians: unit tests

### PR DESCRIPTION
**Pull request by devendra-alt**
# [Math magicians: unit tests](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/project_tests.md)

## 👨‍💻 Requirements Fulfilled
✔️ Completed setup of react testing library
✔️ Used msw API call `__mock__` package
✔️ Used jest snapshot method for checking components being rendered on screen
✔️ Tested `calculator.js` and `operate.js` logic files with Unit testing methods
✔️ Tested component UI for `Header` `Home` `Calculator` `Quote` components
✔️ Tested various user interactions like button clicking and navigating to another page using links 
✔️ Tested calculator interactions
✔️ Tested quote being displayed on the screen after fetching from mock API  
✔️ Added snapshots for applications to test on any changes

### CODE REVIEW

File: src/__tests__/logic.test.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    expect(total).toStrictEqual(null);
+  });
+
+  it('shoud return divide by zero', () => {
+    obj.total = '0';
+    obj.next = '0';
+    const { total } = calculate(obj, '%');
+    expect(total).toStrictEqual('0');
+  });
+});
```
Comment (Strangeal) [inline src/__tests__/logic.test.js]: - [x] Great job on writing the unit test for your project as specified in the requirement! 👏 However, Your calculate test doesn't cover all the calculating logic. Please make sure to write your test to cover all the logic in the `calculate.js` file.

File: src/__tests__/header.test.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    await user.click(screen.getByText(/calculator/i));
+    expect(screen.getByText(/Let's do some math/i)).toBeInTheDocument();
+  });
+  it('should naviaget to quotes', async () => {
+    const user = userEvent.setup();
+    await user.click(screen.getByText(/Quotes/i));
+    await waitForElementToBeRemoved(screen.getByLabelText('loading-msg'));
+    expect(screen.getByRole('button')).toBeInTheDocument();
+  });
+});
```
Comment (Strangeal) [inline src/__tests__/header.test.js]: - [x] Kindly consider taking a second look at your screenshot test as this test is failing. See below for details:
Screenshot 2023-07-26 at 18 28 57

**Review by Strangeal — CHANGES_REQUESTED** [review]
Hi @devendra-alt and @kkmanuu, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implement the features ✅ 
- Nice *UI* design ✅ 
- Pixel Perfect Design ✅ 
- Professional `README` template ✅ 
- Linters are green ✅ 
- Correct `GitHub` workflow ✅ 
- Meaningful **Github** commits ✅ 
- Correct `.github/workflows` for linters ✅ 
- Followed *best-practices* ✅ 
- Well structured project ✅ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/__tests__/header.test.js (hunk 2 above)
Comment (devendra-alt) [reply src/__tests__/header.test.js]: thank you @Strangeal 👍 for valuable feedback

**Comment by devendra-alt** [comment]
## Requirements fulfilled

✔️ Added test coverage for calculator logic
✔️ Fixed quote fetch test description function

**Review by V-Blaze — APPROVED** [review]
Hi @devendra-alt  , 

# Approved 🟢 🟢 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy

## Highlights

- All tests are passing💯
-  Test covers all components and calculator functions👍
-  Great communication with the reviewer🟢 👍

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**